### PR TITLE
BUDI-8084 - Single attachment binding as subtype

### PR DESCRIPTION
**Pull request by adrinr**
## Description
Changing single attachments to have a subtype in order to be cleaner on its checks, not relying only on the max length only. This will not be used yet, but it might be used on bindings and other places. Having this subtype single can be extended in the future, as we can have single/multiple in multiple types, such as files, users, relationships, etc.

This change made me fix some types around the utils, getting a cleaner and safer usage of it

## Feature branch env
[Feature Branch Link](http://fb-budi-8084-single-attachment-binding.fb.qa.budibase.net)

### CODE REVIEW

File: packages/server/src/api/controllers/table/utils.ts [hunk 1]
```diff
@@ -88,6 +89,24 @@ export async function checkForColumnUpdates(
     // Update views
     await checkForViewUpdates(updatedTable, deletedColumns, columnRename)
   }
+
+  for (const attachmentColumn of Object.values(updatedTable.schema).filter(
+    column =>
+      column.type === FieldType.ATTACHMENT &&
+      column.subtype !== oldTable?.schema[column.name]?.subtype
+  )) {
```
Comment (samwho) [inline packages/server/src/api/controllers/table/utils.ts:97]: This is a little bit gnarly, might be a bit easier to read if it's split out into its own variable:

```typescript
const changedAttachmentColumns = Object.values(updatedTable.schema).filter(
  column =>
    column.type === FieldType.ATTACHMENT &&
    column.subtype !== oldTable?.schema[column.name]?.subtype
)
for (const attachmentColumn if changedAttachmentColumns) {
  // ...
```

**Review by samwho — APPROVED** [review]
Really nice type improvements 🚀

File: packages/builder/src/components/backend/DataTable/modals/CreateEditColumn.svelte [hunk 2]
```diff
@@ -191,8 +196,10 @@
     // don't make field IDs for auto types
     if (type === AUTO_TYPE || autocolumn) {
       return type.toUpperCase()
-    } else {
+    } else if (type === FieldType.BB_REFERENCE) {
```
Comment (mike12345567) [inline packages/builder/src/components/backend/DataTable/modals/CreateEditColumn.svelte:199]: should we just check here if it has a `subtype` and do this? Just to cover all sub-type supporting columns?

Comment (adrinr) [reply packages/builder/src/components/backend/DataTable/modals/CreateEditColumn.svelte:199]: This was a 'hack' that was added for bb reference. This assumes that these columns' subtypes, and the old attachments have not

Comment (mike12345567) [reply packages/builder/src/components/backend/DataTable/modals/CreateEditColumn.svelte:199]: Ah I see - makes sense!

**Review by mike12345567 — COMMENTED** [review]
Forgot to click send my approval, but looks good :)

**Pull request merged**